server/services/MonitorService.js

```diff
@@ -137,7 +238,10 @@ export default class MonitorService {
       }
 
       const { callAsCurrentUser } = await this.esDriver.asScoped(req);
-      const updateResponse = await callAsCurrentUser('alerting.updateMonitor', params);
+      const updateResponse = await callAsCurrentUser(
+        `alerting.${type === 'workflow' ? 'updateWorkflow' : 'updateMonitor'}`,
```

Comment:
Non-blocking nitpick: It may be worthwhile to compile an exhaustive list of `type`s in a constant somewhere.